libs/auth/src/java_card.ts

```diff
@@ -318,7 +328,6 @@ export class JavaCard implements Card {
   }
 
   async clearData(): Promise<void> {
-    await this.selectApplet();
```

Comment:
Ah yes good q - should've left a note clarifying!

Because `this.clearData` just calls `this.writeData`, and `this.writeData` calls `this.selectApplet` internally, there's no need to call it here. I was initially worried about someone forgetting to add a `this.selectApplet` back in if they ever updated `this.clearData` to no longer call `this.writeData`, but we know have tests to prevent a regression like this.

All that said, a comment does feel in order since this is the only top-level method that doesn't call `this.selectApplet`